### PR TITLE
Remove unused type

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs [hunk 1]
```diff
@@ -138,7 +137,6 @@ protected virtual void GetServices()
         this.EditorOptionsService = this.Package.ComponentModel.GetService<EditorOptionsService>();
         this.Workspace = this.Package.ComponentModel.GetService<VisualStudioWorkspaceImpl>();
         this.EditorAdaptersFactoryService = this.Package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>();
-        this.AnalyzerFileWatcherService = this.Package.ComponentModel.GetService<AnalyzerFileWatcherService>();
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs:141]: we would create the instance her.e  HOwever:

1. nothing ever read this feild in this type. 
2. the constructor for that type does nothing.

File: src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs [hunk 2]
```diff
@@ -1,110 +0,0 @@
...
-using Microsoft.VisualStudio.Shell.Interop;
-
-namespace Microsoft.VisualStudio.LanguageServices.Implementation;
-
-[Export(typeof(AnalyzerFileWatcherService))]
-[method: ImportingConstructor]
-[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class AnalyzerFileWatcherService(SVsServiceProvider serviceProvider)
-{
-    private readonly IVsFileChangeEx _fileChangeService = (IVsFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs:21]: the constructor of this type does nothing except grab a service, which it places in a field, but never uses.

File: src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs [hunk 3]
```diff
@@ -1,110 +0,0 @@
...
-        {
-            return null;
-        }
-        catch (UnauthorizedAccessException)
-        {
-            return null;
-        }
-    }
-
-    internal void TrackFilePathAndReportErrorIfChanged(string filePath)
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs:52]: this is the only thing that could cause this type to do anything interesting.  but it has no callers in roslyn.  so this entire type is pointless.

removing.  if someone can find a scenario where this is needed, we can return it.  but we've clearly never noticed that it was doing nothing up till now.

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs:141]: i noticed this as i wanted to move these calls into the constructor of AbstractLangService, and i wanted to make sure none of them were doing anything wonky.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**